com.unity.visualeffectgraph/Editor/ShaderGraph/VFXShaderGraphParticleOutput.cs

```diff
@@ -341,24 +341,25 @@ public override VFXExpressionMapper GetExpressionMapper(VFXDeviceTarget target)
                     var shaderGraph = GetOrRefreshShaderGraphObject();
                     if (shaderGraph != null)
                     {
-                        foreach (var tex in shaderGraph.textureInfos.Where(t => t.texture != null).OrderBy(t => t.name))
+                        foreach (var tex in shaderGraph.textureInfos)
                         {
-                            switch (tex.texture.dimension)
+                            switch (tex.dimension)
                             {
+                                default:
                                 case TextureDimension.Tex2D:
-                                    mapper.AddExpression(new VFXTexture2DValue(tex.texture.GetInstanceID(), VFXValue.Mode.Variable), tex.name, -1);
+                                    mapper.AddExpression(new VFXTexture2DValue(tex.instanceID, VFXValue.Mode.Variable), tex.name, -1);
```

Comment:
Not needed in this PR but I wonder why Mode is set to variable. This is only textures withing shader graph that are not exposed right? So having them as constant will allow potential optimization opportunities on our side (like not resetting them always for instance)

Reply:
Just checked the history and it's been like that since the creation of this file. Maybe there is a reason for it?

Reply:
Don't think there's a reason, but anyway we I'll keep that in mind and we can fix later. At the moment the only optimization it allows is to share slot in property sheet for identical textures. But for SG texture, n texture slots will still be used given how it is done anyway. So the optimization is far from massive.

Reply:
Oh actually maybe the reason was that! because we cannot share SG texture because even if constant they require direct namings and cannot be aliased.